components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -489,6 +491,11 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
     ResetSyncInternal();
   if (contains_only_one_device)
     OnResetSync();
+
+  if (waiting_for_second_device && sync_devices->size() >= 2) {
```

Comment:
how can this ever be true? `sync_devices->size() < 2 && sync_devices->size() >= 2` ?

Reply:
before applying resolved records, device size was 1 and after that, it should be 2 or more depends on what resolved records we receive

Reply:
I see, the code that was collapsed updates the sync_device count?